packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,357 @@
+              Prog. Submit
+            </button>
+            <span> </span>
+            <button
+              title='Click me to validate the form programmatically.'
+              className='btn btn-default'
+              type='button'
+              onClick={() => {
+                const valid = playGroundFormRef.current.validateForm();
+                alert(valid ? 'Form is valid' : 'Form has errors');
+              }}
+            >
+              Prog. Validate
+            </button>
+            <div style={{ marginTop: '5px' }} />
+            <CopyLink shareURL={shareURL} onShare={onShare} />
+          </div>
+          <div className='col-sm-2'>
+            <RawValidatorTest validator={validators[validator]} schema={schema} formData={formData} />
+          </div>
+        </div>
+      </div>
+      <Editors
+        {...{
+          extraErrors,
+          formData,
+          schema,
+          setExtraErrors,
+          setFormData,
+          setSchema,
+          setShareURL,
+          setUiSchema,
+          uiSchema,
+        }}
+      />
+      <div className='col-sm-5'>
+        <ErrorBoundary>
+          {showForm && (
+            <DemoFrame
+              head={
+                <>
+                  <link rel='stylesheet' id='theme' href={stylesheet || ''} />
+                  {theme === 'antd' && (
+                    <div
+                      dangerouslySetInnerHTML={{
+                        // @ts-ignore
+                        __html: document.getElementById('antd-styles-iframe')?.contentDocument.head.innerHTML,
+                      }}
+                    />
+                  )}
+                </>
+              }
+              style={{
+                width: '100%',
+                height: 1000,
+                border: 0,
+              }}
+              theme={theme}
+            >
+              <FormComponent
+                {...templateProps}
+                {...liveSettings}
+                schema={schema}
+                uiSchema={uiSchema}
+                formData={formData}
+                onChange={onFormDataChange}
+                noHtml5Validate={true}
+                onSubmit={({ formData }: { formData: string }, e: any) => {
+                  console.log('submitted formData', formData);
+                  console.log('submit event', e);
+                  window.alert('Form submitted');
+                }}
+                fields={{ geo: GeoPosition }}
+                validator={validators[validator]}
+                onBlur={(id: string, value: string) => console.log(`Touched ${id} with value ${value}`)}
+                onFocus={(id: string, value: string) => console.log(`Focused ${id} with value ${value}`)}
+                onError={log('errors')}
+                ref={playGroundFormRef}
+              />
+            </DemoFrame>
+          )}
+        </ErrorBoundary>
+      </div>
+    </>
+  );
+};
+
+export default Playground;
+
+type EditorsProps = {
```

Comment:
Did you want to extract this to `components` too?

Reply:
If you keep it here, I'd suggest putting it before the `Playground` in the file so it's clear to people maintaining it that is used by `Playground` (i.e. defined before used)